examples/Workflow/WorkflowWebApp/Activities/ReserveInventoryActivity.cs

```diff
@@ -1,32 +1,60 @@
 ﻿namespace WorkflowWebApp.Activities
 {
     using System.Threading.Tasks;
+    using Dapr.Client;
     using Dapr.Workflow;
-
-    record InventoryRequest(string RequestId, string Name, int Quantity);
-    record InventoryResult(bool Success);
+    using WorkflowWebApp.Models;
 
     class ReserveInventoryActivity : WorkflowActivity<InventoryRequest, InventoryResult>
     {
         readonly ILogger logger;
+        readonly DaprClient client;
+        private static readonly string storeName = "statestore";
 
-        public ReserveInventoryActivity(ILoggerFactory loggerFactory)
+        public ReserveInventoryActivity(ILoggerFactory loggerFactory, DaprClient client)
         {
             this.logger = loggerFactory.CreateLogger<ReserveInventoryActivity>();
+            this.client = client;
         }
 
         public override async Task<InventoryResult> RunAsync(WorkflowActivityContext context, InventoryRequest req)
         {
             this.logger.LogInformation(
-                "Reserving inventory: {requestId}, {name}, {quantity}",
+                "Reserving inventory for order {requestId} of {quantity} {name}",
                 req.RequestId,
-                req.Name,
-                req.Quantity);
+                req.Quantity,
+                req.ItemName);
+
+            OrderPayload original;
+            string originalETag;
+
+            // Ensure that the store has items
+            (original, originalETag) = await client.GetStateAndETagAsync<OrderPayload>(storeName, req.ItemName);
+
+            // Catch for the case where the statestore isn't setup
+            if (original == null)
+            {
+                // Not enough paperclips.
+                return new InventoryResult(false, original, originalETag);
```

Comment:
Instead of `original` and `originalETag` here, just do `null`, `null`. The intent is otherwise confusing to the reader.

Reply:
"original" is still needed since it is the response from the statestore. If there are no items, we need to check for it. Furthermore, we use 'original' to see the quantity of items as well as logging. I think we should just rename 'original' to 'orderResponse' or something along those lines